cylc/flow/xtriggers/xrandom.py

```diff
@@ -43,28 +44,37 @@ def xrandom(percent, secs=0, _=None):
     Examples:
         If the percent is zero, it returns that the trigger condition was
         not satisfied, and an empty dictionary.
-        >>> xrandom(0, 0)
-        (False, {})
+
+        .. code-block:: python
```

Comment:
We shouldn't have to do this.

The Napoleon extension should automatically interpret the examples section as a Python code-block.

Reply:
But we aren't using Napoleon in cylc-doc, only Rose.

Suggest adding this into the `extension` list in cylc-doc:

```
'sphinx.ext.napoleon'
```

It's a built-in plugin, but it's not enabled by default, it understands the stricture of Google style docstring (what we use) and formats them more nicely.

Reply:
I think `py:exc reference target not found` means that it is trying to cross-reference the exception, but can't because the exception hasn't been documented.

Normally this wouldn't matter, but we turn on nit-picky mode so that referencing errors cause failures.

One option would be to auto-document exceptions in the reference section. I guess this could have some value.

Reply:
[unrelated] Skimming the Rose config I spotted we're using `sphinx.ext.viewcode` there which is kinda cool.